include/avif/avif.h

```diff
@@ -1130,6 +1130,7 @@ AVIF_API avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output
 
 // Codec-specific, optional "advanced" tuning settings, in the form of string key/value pairs.
 // key must be non-NULL, but passing a NULL value will delete that key, if it exists.
+// Settings will be sent to codec once at the next avifEncoderAddImage() call.
```

Comment:
> ```cpp
> // key must be non-NULL, but passing a NULL value will delete that key, if it exists.
> ```

This is now misleading:
- Consider setting option A to value `X` at frame 0; setting A to `NULL` at frame 1 will not discard the setting on the codec side (no-op).

I suggest either:
- Removing the possibility of passing `NULL` to `avifCodecSpecificOptionsSet()`.
- When `NULL` is passed to `avifCodecSpecificOptionsSet()`, instead of removing the setting from the `csOptions`, keep it as `NULL` so it can be "unset" from the codec.

I would lean toward the second solution but I do not know if it can be done.

_Note: By the way, when looking at the implementation of `avifCodecSpecificOptionsSet()` at head, if `NULL` is passed to `avifCodecSpecificOptionsSet()` for a never set option, it will call `avifStrdup(NULL)` and thus `strlen(NULL)`. This is a bug._

- - -

> ```cpp
> // Settings will be sent to codec once at the next avifEncoderAddImage() call.
> ```

It would not hurt to be slightly more precise, what about

```cpp
// See the codec documentation to know if a setting is permanent or applied only to the next frame.
```

Reply:
> This is now misleading

How about now? I'm feeling it's clear enough, but would like you to check it.

> it can be "unset" from the codec

AOM is not providing unset ability, so probably not easy to implement.

The "Passing NULL to delete" behavior is already documented in comment, so maybe just keep it there?

> This is a bug

Fixed.

Reply:
> How about now? I'm feeling it's clear enough, but would like you to check it.

Good, thanks!

> AOM is not providing unset ability, so probably not easy to implement.

I agree.

> The "Passing NULL to delete" behavior is already documented in comment, so maybe just keep it there?

Adding the word "pending" is enough for me, yes.